compute/zgbmm.c

```diff
@@ -0,0 +1,410 @@
+ *          n >= 0.
+ *
+ * @param[in] k
+ *          The number of columns of the matrix op( A ) and the number of rows
+ *          of the matrix op( B ). k >= 0.
+ *
+ * @param[in] kl
+ *          The width of the band below the diagonal in band matrix A.
+ *
+ * @param[in] ku
+ *          The width of the band above the diagonal in band matrix A.
+ *
+ * @param[in] alpha
+ *          The scalar alpha.
+ *
+ * @param[in] pA
+ *          An lda-by-ka matrix, where ka is k when transa = PlasmaNoTrans,
+ *          and is m otherwise.
+ *
+ * @param[in] lda
+ *          The leading dimension of the array A.
+ *          When transa = PlasmaNoTrans, lda >= max(1,m),
+ *          otherwise, lda >= max(1,k).
+ *
+ * @param[in] pB
+ *          An ldb-by-kb matrix, where kb is n when transb = PlasmaNoTrans,
+ *          and is k otherwise.
+ *
+ * @param[in] ldb
+ *          The leading dimension of the array B.
+ *          When transb = PlasmaNoTrans, ldb >= max(1,k),
+ *          otherwise, ldb >= max(1,n).
+ *
+ * @param[in] beta
+ *          The scalar beta.
+ *
+ * @param[in,out] pC
+ *          An ldc-by-n matrix. On exit, the array is overwritten by the m-by-n
+ *          matrix ( alpha*op( A )*op( B ) + beta*C ).
+ *
+ * @param[in] ldc
+ *          The leading dimension of the array C. ldc >= max(1,m).
+ *
+ *******************************************************************************
+ *
+ * @retval PlasmaSuccess successful exit
+ *
+ *******************************************************************************
+ *
+ * @sa plasma_omp_zgbmm
+ * @sa plasma_cgbmm
+ * @sa plasma_dgbmm
+ * @sa plasma_sgbmm
+ *
+ ******************************************************************************/
+int plasma_zgbmm(plasma_enum_t transa, plasma_enum_t transb,
+                 int m, int n, int k, int kl, int ku,
+                 plasma_complex64_t alpha, plasma_complex64_t *pA, int lda,
+                                           plasma_complex64_t *pB, int ldb,
+                 plasma_complex64_t beta,  plasma_complex64_t *pC, int ldc)
+{
+    // Get PLASMA context.
+    plasma_context_t *plasma = plasma_context_self();
+    if (plasma == NULL) {
+        plasma_error("PLASMA not initialized");
+        return PlasmaErrorNotInitialized;
+    }
+
+    // Check input arguments.
+    if ((transa != PlasmaNoTrans) &&
+        (transa != PlasmaTrans) &&
+        (transa != PlasmaConjTrans)) {
+        plasma_error("illegal value of transa");
+        return -1;
+    }
+    if ((transb != PlasmaNoTrans) &&
+        (transb != PlasmaTrans) &&
+        (transb != PlasmaConjTrans)) {
+        plasma_error("illegal value of transb");
+        return -2;
+    }
+    if (m < 0) {
+        plasma_error("illegal value of m");
+        return -3;
+    }
+    if (n < 0) {
+        plasma_error("illegal value of n");
+        return -4;
+    }
+    if (k < 0) {
+        plasma_error("illegal value of k");
+        return -5;
+    }
+    if (kl < 0) {
```

Comment:
I also agree with comment on PR2: kl should not exceed `(m-1)`. Thankfully this error is usually caught later down the road, but is still a good idea to do now.

I'm not sure if the transpose parameters will make a difference here. transposing switches `m` and `n`, but also `kl` and `ku`. So perhaps considering transpose parameters may not be necessary.